src/pinghelper.cpp

```diff
@@ -49,6 +49,9 @@ void PingHelper::start(const QString& serverIpv4Gateway,
   if (!m_pingSender->isValid()) {
     m_pingSender->deleteLater();
     m_pingSender = new DnsPingSender(m_source, this);
+    if (!((DnsPingSender*)m_pingSender)->start()) {
```

Comment:
My 2 Thoughts: 
Looking at the other pingsenders, they seem to do be ready once the constructor was finished. 
Given dnspingsender inherits pingsender - It should behave the same imho. 
It also should override `PingSender::isValid()`  and the if should use that as the input.

Reply:
I agree, I think if `DnsPingSender` is the only one to require a `start()` method, then it would be much better to refactor that one ping sender to behave consistently with the other ping senders. Specifically, they should all follow the contract that:
- Sockets are created (and bound) in the Constructor if they need to be long-lived.
- Fallibility is of that creation/binding is communicated via `isValid()`

Reply:
I agree that I don't like this structure (initialize, then `start`, then ready to `sendPing`). I was trying to do the lightest fix possible, while not changing much of the code. I'm open to doing a bigger fix here. I agree that we should move all of the `start` code to the initializer, and be better about using `isValid`. Will put up a new draft shortly.

Reply:
Did this here. Additionally, I realized the mac ping sender wasn't using `isValid` properly, so I updated that. (The default implementation is to always return `true`.) Perhaps this will fix some other unexpected behavior.